diff --git a/src/Normalizer/NetworkSettingsNormalizer.php b/src/Normalizer/NetworkSettingsNormalizer.php
--- a/src/Normalizer/NetworkSettingsNormalizer.php
+++ b/src/Normalizer/NetworkSettingsNormalizer.php
@@ -83,6 +83,10 @@ class NetworkSettingsNormalizer implements DenormalizerInterface, NormalizerInte
         if (\array_key_exists('Ports', $data) && $data['Ports'] !== null) {
             $values = new \Docker\API\Runtime\JsonObject();
             foreach ($data['Ports'] as $key => $value) {
+                if (null === $value) {
+                    $values[$key] = null;
+                    continue;
+                }
                 $values_1 = [];
                 foreach ($value as $value_1) {
                     $values_1[] = $this->denormalizer->denormalize($value_1, \Docker\API\Model\PortBinding::class, 'json', $context);
@@ -232,6 +236,10 @@ class NetworkSettingsNormalizer implements DenormalizerInterface, NormalizerInte
         if ($data->isInitialized('ports') && null !== $data->getPorts()) {
             $values = new \Docker\API\Runtime\JsonObject();
             foreach ($data->getPorts() as $key => $value) {
+                if (null === $value) {
+                    $values[$key] = null;
+                    continue;
+                }
                 $values_1 = [];
                 foreach ($value as $value_1) {
                     $values_1[] = $value_1 === null ? null : new \Docker\API\Runtime\JsonObject($this->normalizer->normalize($value_1, 'json', $context));
